src/Analyzers/CSharp/CodeFixes/UseUTF8StringLiteral/UseUTF8StringLiteralCodeFixProvider.cs

```diff
@@ -0,0 +1,125 @@
+                nameof(CSharpAnalyzersResources.Use_UTF8_string_literal)),
+                context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            foreach (var diagnostic in diagnostics)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                var arrayNode = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+                var stringValue = diagnostic.Properties[UseUTF8StringLiteralDiagnosticAnalyzer.StringValuePropertyName]!;
+
+                // If we're replacing a byte array that is passed to a parameter array, not and an explicit array creation
+                // then we'll get then arrayNode will be the ArgumentListSyntax so we have to work a bit harder
+                //
+                // eg given a method:
+                //     M(string x, params byte[] b)
+                // our diagnositic would be reported on:
+                //     M("hi", [|1, 2, 3, 4|]);
+                //
+                // but arrayNode will be the whole argument list syntax
+
+                if (arrayNode is ArgumentListSyntax argumentList)
+                {
+                    editor.ReplaceNode(arrayNode, CreateArgumentListWithUTF8String(argumentList, diagnostic.Location, stringValue));
+                }
+                else
+                {
+                    editor.ReplaceNode(arrayNode, CreateUTF8String(arrayNode, stringValue));
+                }
+            }
+
+            return Task.CompletedTask;
+        }
+
+        private static SyntaxNode CreateArgumentListWithUTF8String(ArgumentListSyntax argumentList, Location location, string stringValue)
+        {
+            // To construct our new argument list we add any existing arguments before the location
+            // and then once we hit the location, we add our string literal
+            var _ = ArrayBuilder<ArgumentSyntax>.GetInstance(out var arguments);
+            foreach (var argument in argumentList.Arguments)
+            {
+                if (argument.Span.Start >= location.SourceSpan.Start)
+                {
+                    // If this is the first argument in the argument list, then trivia will be
+                    // attached to the open parentheses, so we don't need to do anything.
+                    var leadingTrivia = argument == argumentList.Arguments[0]
+                        ? SyntaxTriviaList.Empty
+                        : SyntaxFactory.TriviaList(argument.GetAllPrecedingTriviaToPreviousToken());
+
+                    var stringLiteral = CreateUTF8String(leadingTrivia, stringValue, argumentList.Arguments.Last().GetTrailingTrivia());
+                    arguments.Add(SyntaxFactory.Argument(stringLiteral));
```

Comment:
If anyone can work out why this code somehow introduces a newline in the leading trivia, out of thin air, and only for one of the tests, there might be some money in it for you. I have no idea!!

Reply:
https://github.com/dotnet/roslyn/blob/0c7c0b33f0871fc4308eb2d75d77b87fc9293290/src/Workspaces/Core/Portable/CodeActions/CodeAction.cs#L330-L331

This is where the newline is added. It looks like the produced trivia has `SyntaxAnnotation.ElasticAnnotation` somewhere.

Is `ElasticAnnotation` supposed to add a newline like that? Meaning is the problem that we have `ElasticAnnotation` or that its existence isn't handled correctly?

Anyway, it looks like someone has hit this before and decided to filter out `ElasticAnnotation`:

https://github.com/dotnet/roslyn/blob/0c7c0b33f0871fc4308eb2d75d77b87fc9293290/src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs#L117

Reply:
An alternative, instead of filtering out the `ElasticAnnotation`s, is to not get it added in the first place.

```diff
         private static SyntaxNode CreateArgumentListWithUTF8String(ArgumentListSyntax argumentList, Location location, string stringValue)
         {
             // To construct our new argument list we add any existing arguments before the location
             // and then once we hit the location, we add our string literal
-            var _ = ArrayBuilder<ArgumentSyntax>.GetInstance(out var arguments);
+            var _ = ArrayBuilder<SyntaxNodeOrToken>.GetInstance(out var arguments);
             foreach (var argument in argumentList.Arguments)
             {
                 if (argument.Span.Start >= location.SourceSpan.Start)
                 {
                     // If this is the first argument in the argument list, then trivia will be
                     // attached to the open parentheses, so we don't need to do anything.
                     var leadingTrivia = argument == argumentList.Arguments[0]
                         ? SyntaxTriviaList.Empty
                         : SyntaxFactory.TriviaList(argument.GetAllPrecedingTriviaToPreviousToken());

                     var stringLiteral = CreateUTF8String(leadingTrivia, stringValue, argumentList.Arguments.Last().GetTrailingTrivia());
                     arguments.Add(SyntaxFactory.Argument(stringLiteral));
                     break;
                 }

                 arguments.Add(argument);
+                arguments.Add(SyntaxFactory.Token(SyntaxTriviaList.Empty, SyntaxKind.CommaToken, SyntaxTriviaList.Empty));
             }

             return argumentList.WithArguments(SyntaxFactory.SeparatedList<ArgumentSyntax>(arguments));
         }
```

Reply:
Downside of that is losing the trivia on the existing commas from the other parameters.. which of course I could copy across, but whats the bet that introduces some other issue? lol, will have a play.

Reply:
> Downside of that is losing the trivia on the existing commas from the other parameters

Any test case on that which I can try playing with?

Reply:
Sorry, didn't see this. GitHub mobile does a terrible job of notifications :)

The elastic annotation is added by `SyntaxFactory.Token` which is called from the separated list constructor, so avoiding that by preserving the existing comma tokens solves this, and is much easier to boot.

Reply:
> The elastic annotation is added by `SyntaxFactory.Token`

Yes, that's why I manually crafted a token without elastic trivia.

I think a remaining question (which is definitely out of scope of this PR) is: "Do elastic trivia supposed to add a new line that way?"